src/chains/signals.py

```diff
@@ -0,0 +1,45 @@
+import logging
+import os
+from urllib.parse import urljoin
+
+import requests
+from django.db.models.signals import post_delete, post_save
+from django.dispatch import receiver
+
+from .models import Chain
+
+logger = logging.getLogger(__name__)
+
+
+def setup_session():
+    session = requests.Session()
+    adapter = requests.adapters.HTTPAdapter(max_retries=3)
+    session.mount("http://", adapter)
+    session.mount("https://", adapter)
+    return session
+
+
+session = setup_session()
+
+
+@receiver(post_save, sender=Chain)
+@receiver(post_delete, sender=Chain)
+def on_chain_update(sender, **kwargs):
+    logger.info("Chain update. Triggering CGW webhook")
+    cgw_cluster_url = os.environ.get("CGW_URL")
+    if cgw_cluster_url is None:
+        logger.error("CGW_URL is not set. Skipping hook call")
+        return
+    cgw_flush_token = os.environ.get("CGW_FLUSH_TOKEN")
+    if cgw_flush_token is None:
+        logger.error("CGW_FLUSH_TOKEN is not set. Skipping hook call")
+        return
+
+    url = urljoin(cgw_cluster_url, f"/v1/flush/{cgw_flush_token}")
+    try:
+        post = session.post(
+            url, json={"invalidate": "Chains"}
+        )  # TODO: if this throws it might leak the token to the logger
```

Comment:
Since the token is part of the URL I wonder if that happens anyways with our tracking and this is a non-issue

Reply:
Sorry, I don't understand what you mean 😅

Reply:
If the `POST` call is unsuccessful then this is going to log an exception like:


>[WARNING] [MainProcess] Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.HTTPConnection object at 0x107580610>: Failed to establish a new connection: [Errno 61] Connection refused')': /v1/flush/some_random_token

Here `some_random_token` is present in the logs (in production this would be the real token). But since it's also part of the path it'd be already part of what is sent to Kibana 😅

Reply:
Yeah, I think it's not a problem 👍🏻